src/rfkill/rfkill.c

```diff
@@ -257,35 +268,93 @@ static int load_state(
+
+        struct write_queue_item *item, *tmp;
+
+        LIST_FOREACH_SAFE(queue, item, tmp, write_queue) {
+                if ((state_file && strcmp(item->file, state_file) == 0) || idx == item->rfkill_idx) {
+                        log_debug("Canceled previous save state of '%s' to %s.", one_zero(item->state), item->file);
+                        LIST_REMOVE(queue, write_queue, item);
+                        free(item->file);
+                        mfree(item);
+                }
+        }
+}
+
+
+static int save_state_queue(
                 int rfkill_fd,
                 struct udev *udev,
                 const struct rfkill_event *event) {
 
-        _cleanup_udev_device_unref_ struct udev_device *device = NULL;
         _cleanup_free_ char *state_file = NULL;
+        struct write_queue_item *item;
         int r;
 
         assert(rfkill_fd >= 0);
         assert(udev);
         assert(event);
 
-        r = find_device(udev, event, &device);
+        r = determine_state_file(udev, event, &state_file);
+        save_state_queue_remove(event->idx, state_file);
         if (r < 0)
                 return r;
 
-        r = determine_state_file(udev, event, device, &state_file);
-        if (r < 0)
-                return r;
+        item = new0(struct write_queue_item, 1);
+        if (!item)
+                return -ENOMEM;
+
+        item->file = state_file;
+        item->rfkill_idx = event->idx;
+        item->state = event->soft;
+        state_file = NULL;
 
-        r = write_string_file(state_file, one_zero(event->soft), WRITE_STRING_FILE_CREATE|WRITE_STRING_FILE_ATOMIC);
+        LIST_APPEND(queue, write_queue, item);
+
+        return 0;
+}
+
+static int save_state_cancel(
+                int rfkill_fd,
+                struct udev *udev,
+                const struct rfkill_event *event) {
+
+        _cleanup_free_ char *state_file = NULL;
+        int r;
+
+        assert(rfkill_fd >= 0);
+        assert(udev);
+        assert(event);
+
+        r = determine_state_file(udev, event, &state_file);
+        save_state_queue_remove(event->idx, state_file);
         if (r < 0)
-                return log_error_errno(r, "Failed to write state file %s: %m", state_file);
+                return r;
 
-        log_debug("Saved state '%s' to %s.", one_zero(event->soft), state_file);
         return 0;
 }
 
+static int save_state_write(void) {
+        struct write_queue_item *item, *tmp;
+        int result = 0;
+        int r;
+
+        LIST_FOREACH_SAFE(queue, item, tmp, write_queue) {
+                r = write_string_file(item->file, one_zero(item->state), WRITE_STRING_FILE_CREATE|WRITE_STRING_FILE_ATOMIC);
+                if (r < 0) {
+                        result = r;
+                        log_error_errno(r, "Failed to write state file %s: %m", item->file);
+                }
+
+                log_debug("Saved state '%s' to %s.", one_zero(item->state), item->file);
```

Comment:
this log line should probably only be shown, if the if block above doesn't apply.. i.e. you probably want an else line there